src/main/java/com/coreos/jetcd/EtcdClient.java

```diff
@@ -33,4 +75,47 @@ public EtcdClient(ManagedChannelBuilder<?> channelBuilder, EtcdClientBuilder bui
     public EtcdKV newKVClient(KVGrpc.KVFutureStub stub) {
         return new EtcdKVImpl(stub);
     }
+
+    private EtcdAuth newAuthClient(AuthGrpc.AuthFutureStub stub){
+        return new EtcdAuthImpl(stub);
+    }
+
+    protected EtcdAuth getAuthClient(){
+        return authClient;
+    }
+
+    protected EtcdKV getKVClient(){
+        return kvClient;
+    }
+
+    /**
+     * get token from etcd with name and password
+     *
+     * @param channel channel to etcd
+     * @param name auth name
+     * @param password auth password
+     * @return token string
+     * @throws  ConnectException This may be caused as network reason, wrong address
+     * @throws  AuthFailedException This may be caused as wrong username or password
+     */
+    private String authenticate(ManagedChannel channel, ByteString name, ByteString password) throws ConnectException, AuthFailedException {
```

Comment:
Can we split this function into two? One is authenticate returning auth response; the other is get token (maybe we don't even need second func).